packages/sdk/src/credential/m365TenantCredential.ts

```diff
@@ -52,12 +53,20 @@ export class M365TenantCredential implements TokenCredential {
       authorityHost: config.authorityHost,
     };
 
-    this.clientSecretCredential = new ClientSecretCredential(
-      config.tenantId!,
-      config.clientId!,
-      config.clientSecret!,
-      tokenCredentialOptions
-    );
+    if (config.clientSecret) {
```

Comment:
So if both exists in config we will use secret?

Reply:
Yes. 

BTW, this PR is not ready for review yet. I just want to run the CI.
Will close this. and open PR when ready.